caikit_nlp/modules/text_generation/peft_tgis_remote.py

```diff
@@ -107,6 +110,14 @@ def __del__(self):
             if tgis_backend and prompt_cache_id and model_id:
                 tgis_backend.unload_prompt_artifacts(model_id, prompt_cache_id)
 
+    @cached_property
+    def _client(self):
+        # Configure the internal client
+        # NOTE: This is made optional for the cases where we do not need to execute `.run` function
+        # for example, bootstrapping a model to caikit format and saving.
+        if hasattr(self, "tgis_backend") and self._tgis_backend:
+            return self._tgis_backend.get_client(self.base_model_name)
```

Comment:
Added a `_tgis_backend` property which is guaranteed to exist. Allows us to do a simple `if self._tgis_backend` instead of using `getattr()` which breaks intellisense/autocomplete in IDEs.